GitUI/CommandsDialogs/FormPull.cs

```diff
@@ -1163,6 +1164,17 @@ private void PruneTags_CheckedChanged(object sender, EventArgs e)
             AllTags.Checked = AllTags.Checked || PruneTags.Checked;
         }
 
+        private void DisableMergeOptionsWhenCurrentBranchIsLocal()
+        {
+            var isLocalBranch = string.IsNullOrEmpty(Module.GetRemoteBranch(localBranch.Text));
+
+            Rebase.Enabled =
+                Merge.Enabled =
+                Merge.Checked = !isLocalBranch;
```

Comment:
I am not very familiar with this dialog because I do not need/use it.
There is an `AppSettings.DefaultPullAction` to remember the last pull/fetch action.
Please double check that it is not overridden by this piece of code.

Reply:
It seems these things are different, I'm just setting button states, that's it.